scripts/ingest-tarball.sh

```diff
@@ -24,11 +31,137 @@ function echo_red() {
     echo -e "\e[31m$1\e[0m"
 }
 
+function echo_yellow() {
+    echo -e "\e[33m$1\e[0m"
+}
+
 function error() {
     echo_red "ERROR: $1" >&2
     exit 1
 }
 
+function check_version() {
+    # Check if the EESSI version number encoded in the filename
+    # is valid, i.e. matches the format YYYY.DD
+    if ! echo "${version}" | egrep -q '^20[0-9][0-9]\.(0[0-9]|1[0-2])$'
+    then
+        error "${version} is not a valid EESSI version."
+    fi
+
+    # Check if the version encoded in the filename matches the top-level dir inside the tarball
+    if [ "${version}" != "${tar_top_level_dir}" ]
+    then
+        error "the version in the filename (${version}) does not match the top-level directory in the tarball (${tar_top_level_dir})."
+    fi
+}
+
+function check_contents_type() {
+    # Check if the name of the second-level dir in the tarball matches to what is specified in the filename
+    if [ "${contents_type_dir}" != "${tar_contents_type_dir}" ]
+    then
+        error "the contents type in the filename (${contents_type_dir}) does not match the contents type in the tarball (${tar_contents_type_dir})."
+    fi
+
+    # Check if the second-level dir in the tarball is compat, software, or init
+    if [ ! -v content_types[${tar_contents_type_dir}] ]
+    then
+        error "the second directory level of the tarball contents should be either compat, software, or init."
+    fi
+}
+
+function cvmfs_ingest_tarball() {
```

Comment:
I've moved it to a separate function, is that okay with you as well?

Reply:
Well, it's still not clear that `cvmfs_ingest_tarball` also re-generates the catalog...

Reply:
That's something that should be done for every CVMFS transaction/ingestion (i.e. every change in the repo), but somehow isn't done automatically by `cvmfs_server ingest` (while `cvmfs_server publish` does do this). As it's just part of what needs to be done when ingesting a tarball, we don't need to specifically reflect that in the name? But otherwise I propose to move that functionality back into `cvmfs_ingest_tarball` and use the long and ugly name ;-)

Reply:
OK, let's leave it as is, `ingest` without re-generating the catalog doesn't make much sense, indeed...